app/services/search_service.rb

```diff
@@ -44,10 +44,18 @@ def textSearch_all(srchString)
   end
 
   # Search profiles for matching text and package up as a DocResult
-  def textSearch_profiles(srchString)
+  # If order == "recentdesc", search for more recently updated profiles for matching name.
+  # Otherwise, show profiles without any ordering
+  def textSearch_profiles(srchString, order = nil)
     sresult = DocList.new
 
-    users = SrchScope.find_users(srchString, limit = nil)
+    users =
+      if order == "recentdesc"
+        filterMatchingUserName(getRecentProfiles, srchString)
```

Comment:
Here, I worry we might not get any results if we filter after selecting the most recent 100. Is there another way to do this, like finding matching usernames first, sorting by last activity using the user table -- i think there's a last activity column, but not sure when we update that...

Reply:
It might take some kind of clever `join` and `group` query, like, joining revisions to the node table, and sorting by the revision timestamp...